### ISSUE TITLE
Client not syncing to server if counter 3 updated before counter 2 on Personal Quest 588

### ISSUE THREAD

**Issue by Triple904** [Lurkars/gloomhavensecretariat#495]
### Describe the bug

After finishing a session, I went back to GHS after a pause, and our game was at a previous state.  When I went into the menu and clicked 'Undo server sync", we were restored to where we were at the end of our session, and the client was showing a "Missing states -1" message in red at the top.  If I hit "Redo Server Sync", we are back to where we were much earlier in the session, so it would seem the client is ahead of the server and is not syncing.

![image](https://github.com/Lurkars/gloomhavensecretariat/assets/65728662/cb19304a-2872-4621-883d-beea343d18f1)

### Game Enviroment

Frosthaven, Ancient Spire scenario train (Scenario 15).  We clicked continue with scenario 24 and completed that one to end our session.  I am running my own server via docker on an instance in AWS.

### Expected behavior

I would expect the client to upload latest data to server periodically.

### Gloomhaven Secretariat Version

0.83.3

### Used Browser

Installed Client 0.83.3

### OS

MacOSX Sonoma for Client

**Comment by Triple904** [comment]
I brought forward the state by basically manually completing the scenario and updating character attributes (exp, items, etc.).  It seemed to save it fine, and I was able to open on a new client (different machine) with updated state.

However, if I go into the character sheet and update the personal quest (in this case, Robotic Boltshooters killed):

![image](https://github.com/Lurkars/gloomhavensecretariat/assets/65728662/480c8b9f-a074-4faa-8663-0f96096763da)

It seems to show the server as synced.  The undos look like this:

![image](https://github.com/Lurkars/gloomhavensecretariat/assets/65728662/4bea8dd8-c41e-4853-81e9-229bd7d9a66d)

If I now close the client, and re-open it, the personal quest is no longer updated:

![image](https://github.com/Lurkars/gloomhavensecretariat/assets/65728662/54e7f89f-cb7c-409b-b695-d735d598b4e6)

And the server indicates "Missing states":

![image](https://github.com/Lurkars/gloomhavensecretariat/assets/65728662/dbddfa8c-5f77-4b8c-a0a8-27c68fab6fb8)

**Comment by Triple904** [comment]
I have also updated the server docker image to 0.84.11 and client to 0.84.12, but this made no difference.

**Comment by Triple904** [comment]
OK, after further mucking about, this seems to have something to do with updating this personal quest.  If I update counter 3 (Robotic Boltshooters) before updating counter 2 (Lurker Clawcrushers), I get the above syncing errors.  If however, I update counter 2 by checking even a single Clawcrusher box, I can then check both Boltshooter boxes and the server will sync.  If I then uncheck a Clawcrusher box, syncing still works.

During our original session, I believe I updated the Boltshooters on the Deathwalker early on, and that is why we ran into the initial problem of returning to an early state after finishing our session.  Updating bug title to reflect findings.
